pkg/storage/etcd3/watcher.go

```diff
@@ -158,13 +158,17 @@ func (wc *watchChan) sync() error {
 	wc.initialRev = getResp.Header.Revision
 
 	for _, kv := range getResp.Kvs {
-		prevResp, err := wc.watcher.client.Get(wc.ctx, string(kv.Key), clientv3.WithRev(kv.ModRevision-1), clientv3.WithSerializable())
-		if err != nil {
-			return err
-		}
 		var prevVal []byte
-		if len(prevResp.Kvs) > 0 {
-			prevVal = prevResp.Kvs[0].Value
+		if kv.ModRevision != kv.CreateRevision {
+			prevResp, err := wc.watcher.client.Get(wc.ctx, string(kv.Key), clientv3.WithRev(kv.ModRevision-1), clientv3.WithSerializable())
+			if err != etcdrpc.ErrCompacted {
+				if err != nil {
+					return err
+				}
+				if len(prevResp.Kvs) > 0 {
+					prevVal = prevResp.Kvs[0].Value
+				}
+			}
```

Comment:
etcd2 does not always send Add... see https://github.com/kubernetes/kubernetes/issues/36545#issuecomment-259613494